pinecone/client.go

```diff
@@ -421,39 +460,82 @@ func derefOrDefault[T any](ptr *T, defaultValue T) T {
 	return *ptr
 }
 
-func buildClientOptions(in NewClientParams) ([]control.ClientOption, error) {
+func (ncp *NewClientParams) buildClientOptions() ([]control.ClientOption, error) {
```

Comment:
could buildClientOptions to be a wrapper around buildClientBaseOptions?

Reply:
I slimmed things down to just `buildClientBaseOptions`, lemme know if that makes sense.